### PR TITLE
Add default impl to getToolTypeName and add a null check

### PR DESCRIPTION
**Pull request by Caedis**
Related to https://github.com/GTNewHorizons/GT5-Unofficial/pull/2423

### CODE REVIEW

**Comment by LewisSaber** [comment]
Thatnks, I wanted to fix it in other mods this morning, didnt know you can add default implementation, can you also implement getMaxMode, thanks!

File: src/main/java/gregtech/api/interfaces/IToolStats.java [hunk 1]
```diff
@@ -195,7 +195,9 @@ int convertBlockDrops(List<ItemStack> aDrops, ItemStack aStack, EntityPlayer aPl
     float getMiningSpeed(Block aBlock, byte aMetaData, float aDefault, EntityPlayer aPlayer, World worldObj, int aX,
         int aY, int aZ);
 
-    String getToolTypeName();
+    default String getToolTypeName() {
+        return null;
+    };
 
     byte getMaxMode();
```
Comment (LewisSaber) [inline src/main/java/gregtech/api/interfaces/IToolStats.java:202]: This need default implementatino too

**Review by LewisSaber — CHANGES_REQUESTED** [review]

**Comment by Caedis** [comment]
Default was added in java 8 surprisingly

**Review by LewisSaber — APPROVED** [review]
Thanks again!

**Pull request merged**